flink-connectors/flink-connector-hbase/src/main/java/org/apache/flink/connector/hbase/source/AbstractTableInputFormat.java

```diff
@@ -68,6 +71,11 @@ public AbstractTableInputFormat(org.apache.hadoop.conf.Configuration hConf) {
 		serializedConfig = HBaseConfigurationUtil.serializeConfiguration(hConf);
 	}
 
+	/**
+	 * Creates a {@link Scan} object and opens the {@link HTable} connection to initialize the HBase table.
+	 */
+	protected abstract void initTable();
```

Comment:
If you make this method `throws IOException`, you don't need to wrap the IOExceptions in HBaseRowDataInputFormat.connectToTable() into RuntimeExceptions.

Throwing an IOException is not a problem, because initTable is called in open(); which throws an IOException as well.

Reply:
@rmetzger Good idea. I have followed up with your point to `throws IOException` for `initTable()`.